app/models/user.rb

```diff
@@ -0,0 +1,34 @@
+require 'bcrypt'
+
+class User
+
+  attr_reader :password
+  attr_accessor :password_confirmation, :user_handle
+
+  include DataMapper::Resource
+
+  property :id, Serial
+  property :email, String, unique: true, message: 'This email is already taken'
+  property :handle, String, unique: true,
+            message: 'This user name is already taken'
+  property :password_digest, Text
+
+  def password=(password)
+    @password = password
+    self.password_digest = BCrypt::Password.create(password)
+  end
+
+  validates_uniqueness_of :email, :handle
```

Comment:
Prefer the new style validations `validates :column, uniqueness: value` over `validates_uniqueness_of`.